openc3/lib/openc3/accessors/binary_accessor.rb

```diff
@@ -108,13 +106,171 @@ def self.raise_buffer_error(read_write, buffer, data_type, given_bit_offset, giv
+          current_bit_size = 30
+        else
+          current_bit_size = 62
+        end
+
+        if item.data_type == :UINT
+          if value <= 63
+            # Length = 0, value up to 6-bits
+            new_bit_size = 6
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 0)
+          elsif value <= 16383
+            # Length = 1, value up to 14-bits
+            new_bit_size = 14
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 1)
+          elsif value <= 1073741823
+            # Length = 2, value up to 30-bits
+            new_bit_size = 30
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 2)
+          else
+            # Length = 3, value up to 62-bits
+            new_bit_size = 62
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 3)
+          end
+        else
+          if value <= 31 and value >= -32
+            # Length = 0, value up to 6-bits
+            new_bit_size = 6
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 0)
+          elsif value <= 8191 and value >= -8192
+            # Length = 1, value up to 14-bits
+            new_bit_size = 14
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 1)
+          elsif value <= 536870911 and value >= -536870912
+            # Length = 2, value up to 30-bits
+            new_bit_size = 30
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 2)
+          else
+            # Length = 3, value up to 62-bits
+            new_bit_size = 62
+            item.bit_size = new_bit_size
+            @packet.write(item.variable_bit_size['length_item_name'], 3)
+          end
+        end
+
+        # Later items need their bit_offset adjusted by the change in this item
+        adjustment = new_bit_size - current_bit_size
+        bytes = (adjustment / 8)
+        item_offset = item.bit_offset / 8
+        if bytes > 0
+          original_length = buffer.length
+          # Add extra bytes because we're adjusting larger
+          buffer << ("\000" * bytes)
+          # We added bytes to the end so now we have to shift the buffer over
+          #   NOTE: buffer[offset, length]
+          # We copy to the shifted offset location with the remaining buffer length
+          buffer[item_offset + bytes, buffer.length - (item_offset + bytes)] =
+              # We copy from the original offset location with the original length minus the offset
+              buffer[item_offset, original_length - item_offset]
+        elsif bytes < 0
+          # Remove extra bytes because we're adjusting smaller
+          buffer[item_offset + -bytes, -bytes] = ''
```

Comment:
This doesn't seem right.  Let's say we got 7 bytes smaller.   
buffers[0 + 7, 7] = "" will destroy a bunch of real data.
Should it always be: buffer[item_offset + 1, -bytes] = '' ?